Logstreamer crash if missing file_match config

The LogstreamerInput plugin crashes if I omit the file_match config.
If I uncomment the line in the config below, it loads up fine.

``` toml
[json_log]
type = "LogstreamerInput"
log_directory = "/var/log/json"
decoder = "JsonDecoder"
#file_match = '.*\.log\.?(?P<Index>\d+)?(\.gz)?'
priority = ["^Index"]
```

```
panic: runtime error: index out of range

goroutine 16 [running]:
runtime.panic(0x98b0e0, 0xef521c)
    /usr/local/go/src/pkg/runtime/panic.c:279 +0xf5
github.com/mozilla-services/heka/logstreamer.fileMatchRegexp(0xc2080a82d0, 0xd, 0x0, 0x0, 0x0)
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/logstreamer/filehandling.go:379 +0x14a
github.com/mozilla-services/heka/logstreamer.NewLogstreamSet(0xc20807a9b0, 0x9356907420000, 0xc208036810, 0xd, 0xc20809b940, 0x1c, 0x438d2d, 0x0, 0x0)
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/logstreamer/filehandling.go:410 +0x340
github.com/mozilla-services/heka/plugins/logstreamer.(*LogstreamerInput).Init(0xc208058780, 0x84f6c0, 0xc2080aa000, 0x0, 0x0)
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/plugins/logstreamer/logstreamer_input.go:158 +0x6c8
github.com/mozilla-services/heka/pipeline.(*PipelineConfig).loadSection(0xc20803e000, 0xc20804e840, 0x0, 0x0)
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/config.go:622 +0x3fd
github.com/mozilla-services/heka/pipeline.(*PipelineConfig).LoadFromConfigFile(0xc20803e000, 0x7fffbbdfd93a, 0x15, 0x0, 0x0)
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/pipeline/config.go:937 +0x1180
main.loadFullConfig(0xc20803e000, 0xc208037060, 0x0, 0x0)
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/cmd/hekad/main.go:215 +0x5c2
main.main()
    /home/admin/heka/build/heka/src/github.com/mozilla-services/heka/cmd/hekad/main.go:185 +0xe43
```

--- a/CHANGES.txt
+++ b/CHANGES.txt
@@ -4,6 +4,8 @@
 Bug Handling
 ------------
 
+* Fail gracefully on LogstreamerInput missing `file_match` setting (#1105).
+
 * Fixed issue where ProcessInput wasn't propagating environment variables and
   working directory to repeatedly run (vs. one time, long running) processes
   (#1108).

--- a/plugins/logstreamer/logstreamer_input.go
+++ b/plugins/logstreamer/logstreamer_input.go
@@ -16,6 +16,7 @@ package logstreamer
 
 import (
 	"code.google.com/p/go-uuid/uuid"
+	"errors"
 	"fmt"
 	ls "github.com/mozilla-services/heka/logstreamer"
 	"github.com/mozilla-services/heka/message"
@@ -106,6 +107,10 @@ func (li *LogstreamerInput) Init(config interface{}) (err error) {
 	// Setup the journal dir
 	if err = os.MkdirAll(conf.JournalDirectory, 0744); err != nil {
 		return err
+	}
+
+	if conf.FileMatch == "" {
+		return errors.New("`file_match` setting is required.")
 	}
 
 	if len(conf.FileMatch) > 0 && conf.FileMatch[len(conf.FileMatch)-1:] != "$" {

